spock-specs/src/test/resources/snapshots/org/spockframework/smoke/ast/DataTablesAstSpec/filter_block_becomes_its_own_method.groovy

```diff
@@ -0,0 +1,58 @@
+package aPackage
+import spock.lang.*
+
+class ASpec extends Specification {
+  def "aFeature"() {
+/*--------- tag::snapshot[] ---------*/
+public void $spock_feature_0_0(java.lang.Object a, java.lang.Object b) {
+    org.spockframework.runtime.ErrorCollector $spock_errorCollector = org.spockframework.runtime.ErrorRethrower.INSTANCE
+    org.spockframework.runtime.ValueRecorder $spock_valueRecorder = new org.spockframework.runtime.ValueRecorder()
+    try {
+        org.spockframework.runtime.SpockRuntime.verifyCondition($spock_errorCollector, $spock_valueRecorder.reset(), 'true', 2, 7, null, $spock_valueRecorder.record($spock_valueRecorder.startRecordingValue(0), true))
+    }
+    catch (java.lang.Throwable $spock_condition_throwable) {
+        org.spockframework.runtime.SpockRuntime.conditionFailedWithException($spock_errorCollector, $spock_valueRecorder, 'true', 2, 7, null, $spock_condition_throwable)}
+    finally {
+    }
+    this.getSpecificationContext().getMockController().leaveScope()
+}
+
+public java.lang.Object $spock_feature_0_0prov0() {
+    return [1, 2]
+}
+
+public java.lang.Object $spock_feature_0_0prov1(java.util.List $spock_p_a) {
+    return [3]
+}
+
+public java.lang.Object $spock_feature_0_0proc(java.lang.Object $spock_p0, java.lang.Object $spock_p1) {
+    java.lang.Object a = (( $spock_p0 ) as java.lang.Object)
+    java.lang.Object b = (( $spock_p1 ) as java.lang.Object)
+    return new java.lang.Object[]{ a , b }
+}
+
+public static org.spockframework.runtime.model.DataVariableMultiplication[] $spock_feature_0_0prods() {
+    return new org.spockframework.runtime.model.DataVariableMultiplication[]{new org.spockframework.runtime.model.DataVariableMultiplication(new java.lang.String[]{'a', 'b'}, new org.spockframework.runtime.model.DataVariableMultiplicationFactor(new java.lang.String[]{'a'}), new org.spockframework.runtime.model.DataVariableMultiplicationFactor(new java.lang.String[]{'b'}))}
+}
+
+public static void $spock_feature_0_0filter(java.lang.Object a, java.lang.Object b) {
+    org.spockframework.runtime.ErrorCollector $spock_errorCollector = org.spockframework.runtime.ErrorRethrower.INSTANCE
+    org.spockframework.runtime.ValueRecorder $spock_valueRecorder = new org.spockframework.runtime.ValueRecorder()
+    try {
+        org.spockframework.runtime.SpockRuntime.verifyCondition($spock_errorCollector, $spock_valueRecorder.reset(), 'a == 1', 15, 7, null, $spock_valueRecorder.record($spock_valueRecorder.startRecordingValue(2), $spock_valueRecorder.record($spock_valueRecorder.startRecordingValue(0), a) == $spock_valueRecorder.record($spock_valueRecorder.startRecordingValue(1), 1)))
```

Comment:
Just wondering if it makes sense to have this whole infrastructure, when we never show the result. Of course it is easier to re-use the existing rewriter, OTOH it would suffice if we would 

```groovy
return (filterLine1) 
       && (filterLine2)... // and so on if we have multiple lines.
```

Reply:
That's right, but I thought it might be more important to be consistent with the other rewritten code. And also maybe the output gets used in the future.

It could for example be imaginable that we add some "debug filters" switch that prints if an iteration was filtered and why.

Reply:
Without the rewriting, you can for example not use collection conditions, or the special hamcrest matcher methods, and whatever else the rewriting does.

Reply:
Actually, I liked the debug switch option and added it now :-)

Reply:
Combining this with the naming discussion, having simple boolean expressions would allow having both `include`/`exclude`, which could be rendered like

```
return (includeLine1)
       && !(excludeLine1)
```

Doing the same would be more complicated with other conditions. 

I'm not saying we need to switch, but to explore if having both `include`/`exclude` would allow for a more natural usage. We also have both `@IgnoreIf`/`@Requires` even though you could achieve the same with just one.

Reply:
Actually, I like the idea to be able to use all conditions I can also use in other blocks the same way. Writing just a boolean expression statement but with different semantics / edge cases is imho more confusing than helping and will produce several issues when people start to use it.

I also like having the great Spock condition rendering, even if just when enabling logging, which can be quite more helpful than just the boolean expression.

Nevertheless we could of course add an `include` and an `exclude` block and in both support the syntax like now in the `filter` block. But - yes I know I'm exaggerating - we should then also add a `thennot` block and a `dontexpect` block. :-)

I'm not sure whether it was a good idea to add `Requires` when there already was `IgnoreIf` actually. But I personally don't think we need two blocks for this where you then also have to care about precedence and so on.

Imho it is good enough or maybe even better to have
```groovy
filter:
    a == 1
    b != 2
```
instead of
```groovy
include:
    a == 1

exclude:
    b == 2
```
But maybe someone else is adding their opinion here too. :-)

Also the main use-case imho is a bit different.

While we added the possibility to also use data variables in `IgnoreIf` and `Requires` relatively recently, this was not there originally.
The main use-case of those is something like "ingore if running on java 11" or "require Groovy 3 or newer" and similar.

The main use-case of this is to check the data variables for certain constellations just like in `then` or `expect` just with a different consequence. With this `filter` block in place earlier, maybe we even would have never added the possibility to check data variables in `IgnoreIf` and `Requires`.